include/vcml/debugging/gdbserver.h

```diff
@@ -35,18 +35,21 @@ enum gdb_signal {
     GDBSIG_KILL = 9,
 };
 
-struct gdb_target {
-    u64 tid;
-    u64 pid;
-    string xml;
-    vector<const cpureg*> cpuregs;
-    target* tgt;
-};
-
 class gdbserver : public rspserver, private subscriber, private suspender
 {
 private:
-    vector<gdb_target*> m_targets;
+    struct gdb_target {
+        u64 tid;
+        u64 pid;
+        string xml;
+        vector<const cpureg*> cpuregs;
+        target& tgt;
+
+        gdb_target(u64 t, u64 p, const char* x, target& tg):
```

Comment:
Do we need `x`? `xml` will always be initialized to an empty string, so we can just remove `x`  and use the default constructor for `xml`.

Reply:
This is as a placeholder for later when we maybe want to create the xml file in `add_target`. That might be necessary if we want to use the xml file to differentiate between different architectures of the targets.